packages/use-optionally-controlled-state/src/index.tsx

```diff
@@ -1,37 +1,36 @@
 import {useState, useCallback} from 'react';
 import useConstant from 'use-constant';
 
-type Options<Value> =
-  | {
-      controlledValue?: Value;
-      initialValue: Value;
-      onChange?(value: Value): void;
-    }
-  | {
-      controlledValue: Value;
-      initialValue?: Value;
-      onChange?(value: Value): void;
-    };
-
 /**
  * Enables a component state to be either controlled or uncontrolled.
  */
-export default function useOptionallyControlledState<Value>({
+export default function useOptionallyControlledState<
```

Comment:
TODO:
- [ ] Rename to `useOptionalState` as it delivers the same meaning ("controlled" means there's no intermediate state)